src/components/core/compute/initialize.ts

```
@@ -104,14 +110,58 @@ export class ComputeInitializeHandler extends Handler {
               }
             }
           }
+
+          const config = await getConfiguration()
+          const { rpc, network, chainId, fallbackRPCs } =
+            config.supportedNetworks[ddo.chainId]
+          const blockchain = new Blockchain(rpc, network, chainId, fallbackRPCs)
+          const { ready, error } = await blockchain.isNetworkReady()
+          if (!ready) {
+            return {
+              stream: null,
+              status: {
+                httpStatus: 400,
+                error: `Initialize Compute: ${error}`
+              }
+            }
+          }
+
+          const signer = blockchain.getSigner()
+
+          // check if oasis evm or similar
+          const confidentialEVM = isConfidentialChainDDO(ddo.chainId, service)
           // let's see if we can access this asset
           let canDecrypt = false
           try {
-            await decrypt(
-              Uint8Array.from(Buffer.from(sanitizeServiceFiles(service.files), 'hex')),
-              EncryptMethod.ECIES
-            )
-            canDecrypt = true
+            if (!confidentialEVM) {
+              await decrypt(
+                Uint8Array.from(Buffer.from(sanitizeServiceFiles(service.files), 'hex')),
+                EncryptMethod.ECIES
+              )
+              canDecrypt = true
+            } else {
+              // TODO use oasis sdk to decrypt
+              CORE_LOGGER.warn('TODO confidential EVM initialize compute')
```

Let's not put a TODO in the logs. Whenever we have a warning in the logs it leads to a load of questions on Discord. Can you make sure we are also tracking the TODOs in an issue if you don't implement them on this PR.

---

i can throw an "not implemented" exception instead of logging. This is something that is not even defined ATM

---

created an issue to address the signatures for confidential evm c2d